### PR TITLE
The initial docs skeleton

### PR DESCRIPTION
**Pull request by Francis-Gurr**
- Updated the main README.md (moved the existing vuetify one, do we want to keep this or delete it?)
- Added steps 1 through 4
- Step 1 is mostly complete, steps 2-4 need fleshing out
- Added my-books.json and a post request to the api service to fit with the steps

### CODE REVIEW

File: docs/setup.md [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+Open the Mac terminal application and run the following commands (`# these are comments`).
+
+```bash
+# download and install Node.js
+brew install node@20
+
+# verifies the right Node.js version is in the environment
+node -v # should print `v20.15.0`
+
+# verifies the right NPM version is in the environment
```
Comment (SamLowth) [inline docs/setup.md:51]: maybe add a point to what npm is?

File: src/services/api.service.js [hunk 2]
```diff
@@ -1,8 +1,8 @@
 const serverLatency = 0;
 
-export function get (endpoint, payload = {}) {
```
Comment (SamLowth) [inline src/services/api.service.js:3]: worth adding a .md file to explain the api layer / how endpoints work ect. ?

File: docs/tasks/step-3.md [hunk 3]
```diff
@@ -0,0 +1,31 @@
+# Step 3: List all the books in a table
+
+## a) Add table component
+
+On the `/all-books` page list all of the books using a [vuetify data table](https://vuetifyjs.com/en/components/data-tables/basics/).
+
+## c) Add at least one of the following features to the table
+
+### Search
```
Comment (SamLowth) [inline docs/tasks/step-3.md:9]: should these links go to different places e.g.:

https://vuetifyjs.com/en/components/data-tables/data-and-display/#search
https://vuetifyjs.com/en/components/data-tables/data-and-display/#pagination-examples
https://vuetifyjs.com/en/components/data-tables/data-and-display/#filtering-examples

File: docs/tasks/step-2.md
Comment (SamLowth) [inline docs/tasks/step-2.md:1]: just an idea, should we add some images to the task files to show what it should look like at this point? or do we want it more left to imagination?

Comment (Francis-Gurr) [reply docs/tasks/step-2.md:1]: I think leave it up to their imagination? But I think we were also planning to have a branch for each completed step so there would be that for reference

File: docs/tasks/step-3.md (hunk 3 above)
Comment (Francis-Gurr) [reply docs/tasks/step-3.md:9]: Yes they were supposed to, oops, nice spot

File: src/services/api.service.js (hunk 2 above)
Comment (Francis-Gurr) [reply src/services/api.service.js:3]: Yeah that could be good!

File: docs/setup.md (hunk 1 above)
Comment (Francis-Gurr) [reply docs/setup.md:51]: Yeah can do

File: src/services/api.service.js (hunk 2 above)
Comment (Francis-Gurr) [reply src/services/api.service.js:3]: I'll do this in another branch, as I'm planning to add more endpoints

File: docs/setup.md (hunk 1 above)
Comment (Francis-Gurr) [reply docs/setup.md:51]: I'm going to actually leave this for now, we can add more detail later if we think we need it. I'm also wary of not explaining everything, as that could go on forever. Like NPM is the node package manager. What's node, what's a package, what's a package manager, that'll then lead to more questions. I think we can't explain everything and we'll have people there during the workshop/work experience to explain anything else too.

**Review by dev-steverob — APPROVED** [review]
LGTM

**Pull request merged**